File: src/main/java/me/yeeunhong/waitingforbooking/dto/SignUpDto.java
package me.yeeunhong.waitingforbooking.dto;

//import lombok.AllArgsConstructor;
//import lombok.Builder;
//import lombok.Getter;
//import lombok.NoArgsConstructor;
//import me.yeeunhong.waitingforbooking.domain.User;
//
//import java.util.ArrayList;
//import java.util.List;
//
//@Getter
//@NoArgsConstructor
//@AllArgsConstructor
//@Builder
//public class SignUpDto {
//    private String username;
//    private String password;
//    private List<String> roles = new ArrayList<>();
//
//    public User toEntity(String encodedPassword, List<String> roles) {
//        return User.builder()
//                .username(username)
//                .password(encodedPassword)
//                .roles(roles)
//                .build();
//    }
//}

File: src/main/kotlin/me/yeeunhong/waitingforbooking/dto/SignUpDto.kt
package me.yeeunhong.waitingforbooking.dto

import me.yeeunhong.waitingforbooking.domain.User
data class SignUpDto(
    var username: String,
    val password:String,
    val roles: List<String> = ArrayList()
) {
    fun toEntity(encodedPassword: String, roles: List<String>): User {
        return User.builder()
            .username(username)
            .password(encodedPassword)
            .roles(roles)
            .build()
    }
}